pkg/restore/restore.go

```diff
@@ -1435,20 +1435,30 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 					ctx.log.Infof("restore API has resource policy defined %s , executing restore workflow accordingly for changed resource %s %s", resourcePolicy, fromCluster.GroupVersionKind().Kind, kube.NamespaceAndName(fromCluster))
 
 					// existingResourcePolicy is set as none, add warning
-					if resourcePolicy == velerov1api.PolicyTypeNone {
+					switch resourcePolicy {
+					case velerov1api.PolicyTypeNone:
 						e := errors.Errorf("could not restore, %s %q already exists. Warning: the in-cluster version is different than the backed-up version",
 							obj.GetKind(), obj.GetName())
 						warnings.Add(namespace, e)
+					case velerov1api.PolicyTypeUpdate, velerov1api.PolicyTypeRecreate:
 						// existingResourcePolicy is set as update, attempt patch on the resource and add warning if it fails
-					} else if resourcePolicy == velerov1api.PolicyTypeUpdate {
 						// processing update as existingResourcePolicy
 						warningsFromUpdateRP, errsFromUpdateRP := ctx.processUpdateResourcePolicy(fromCluster, fromClusterWithLabels, obj, namespace, resourceClient)
 						if warningsFromUpdateRP.IsEmpty() && errsFromUpdateRP.IsEmpty() {
 							itemStatus.action = itemRestoreResultUpdated
 							ctx.restoredItems[itemKey] = itemStatus
 						}
-						warnings.Merge(&warningsFromUpdateRP)
-						errs.Merge(&errsFromUpdateRP)
+						// if resourcePolicy is recreate, attempt to recreate if patch had errors
+						if !errsFromUpdateRP.IsEmpty() && resourcePolicy == velerov1api.PolicyTypeRecreate{
+							warningsFromRecreateRP, errsFromRecreateRP := ctx.processRecreateResourcePolicy(fromCluster, fromClusterWithLabels, obj, namespace, resourceClient)
+							if warningsFromRecreateRP.IsEmpty() && errsFromRecreateRP.IsEmpty() {
+								itemStatus.action = itemRestoreResultUpdated
+								ctx.restoredItems[itemKey] = itemStatus
+							}
```

Comment:
TODO: merge warningsFromRecreateRP, errsFromRecreateRP

Reply:
resolved in https://github.com/vmware-tanzu/velero/compare/6723627b11f1da5cf9a901f3921439194dd45875..b8ea1c4a6bd4cf968a5207e842e2c8c8089a1127